third_party/httplib/httplib.hpp

```diff
@@ -3293,7 +3299,10 @@ inline bool read_content_with_length(Stream &strm, uint64_t len,
 	uint64_t r = 0;
 	while (r < len) {
 		auto read_len = static_cast<size_t>(len - r);
-		auto n = strm.read(buf, (std::min)(read_len, CPPHTTPLIB_RECV_BUFSIZ));
+		auto n = (uint64_t) strm.read(buf, (std::min)(read_len, CPPHTTPLIB_RECV_BUFSIZ));
```

Comment:
I feel like we would prefer to have this code not as a modification to the inlined library and ideally handle this only from the content receiver callback

Also, is this even necessary? aren't you already handling this in the content receiver already?

Reply:
Yes, I was also not super happy changing the third-party lib.
The nice thing is this can be an early-out in certain situations. Since the to-be-read batch can already be over our limit.

The content receiver still has to process stuff, potentially wasting more work.

Reply:
Ah that's a fair point. I would however still argue that the added code complexity from writing our code in an inlined lib is not worth saving reading the bytes. I mean this should be an edge case that a user will only run when his `get_request_file_size` is too low, right?

Reply:
By default, the `get_request_file_size = 100mb' maybe we should make it smaller and remove it from the third-party lib?
@Mytherin, any opinions here?

Reply:
I agree that editing the httplib is likely not worth the hassle. Perhaps we should just always download the entire file and remove the file size parameter entirely? I also wonder how this PR relates to #6467 - does that solve some of the cases solved by this PR as well?

Reply:
I've ran the same tests here with the #6467 PR, it seems that the range-requests can't cover the missing content-length header.

I've removed the file size requirement entirely.